node/silkworm/snapshot/bittorrent.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+   Copyright 2022 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <filesystem>
+#include <memory>
+#include <set>
+#include <string>
+#include <thread>
+#include <vector>
+
+// Disable warnings raised during compilation of libtorrent
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wc++11-compat"
+#pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#include <libtorrent/session.hpp>
+#include <libtorrent/session_params.hpp>
+#pragma GCC diagnostic pop
+
+namespace silkworm {
+
+constexpr const char* kDefaultTorrentRepoPath{".torrent"};
+constexpr const char* kSessionFileName{".session"};
+constexpr const char* kResumeDirName{".resume"};
+constexpr const char* kResumeFileExt{".resume"};
+constexpr const char* kDefaultMagnetsFilePath{".magnet_links"};
+constexpr std::chrono::milliseconds kDefaultWaitBetweenAlertPolls{100};
+constexpr std::chrono::seconds kDefaultResumeDataSaveInterval{60};
+constexpr bool kDefaultSeeding{false};
+
+constexpr int kDefaultDownloadRateLimit{64 * 1024 * 1024};  // 64MiB
+constexpr int kDefaultUploadRateLimit{4 * 1024 * 1024};     // 4MiB
+constexpr int kDefaultActiveDownloads{6};
+constexpr int kDefaultMaxOutRequestQueue{6000};
+constexpr bool kDefaultAnnounceToAllTiers{true};
+constexpr int kDefaultAsyncIOThreads{32};
```

Comment:
These definitions will appear on the main namespace

Reply:
Moved to `BitTorrentSettings` and `BitTorrentClient` in https://github.com/torquem-ch/silkworm/pull/785/commits/93d942993a47a84a6f70ccc5fbeb68753e4426b3